### ISSUE TITLE
Zero  GPU_mem   box_loss   seg_loss     dfl_loss  Instances during segmentation task

### ISSUE THREAD

**Issue by aymanaboghonim** [ultralytics/ultralytics#6232]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

while I am fine-tuning my custom model on a cutome dataset  during instance segmentation task, I noticed that all training metrics are zero except class loss .
in the attached screenshot, you can find all the output after I initialized the yolo segment training job.
![image](https://github.com/ultralytics/ultralytics/assets/62246708/22219dd4-49ae-4688-85b2-0d65f0f6f975)

### Environment

Ultralytics YOLOv8.0.207 🚀
 Python-3.8.17
 torch-2.1.0+cu121 
CPU (Intel Xeon 2.30GHz)
OS: linux 
Cuda : Cuda12.1
RAM: 60 GB

### Minimal Reproducible Example

yolo segment train data=data.yml model=best.pt epochs=100 imgsz=640

### Additional

I used yolo on the old version of the dataset and it worked as expected in the same env / settings .
here is a screenshot of my data.yml
![image](https://github.com/ultralytics/ultralytics/assets/62246708/ae03089f-222d-4ce8-8466-152f9350da70)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by aymanaboghonim** [comment]
I noticed that buildings (instances) do not appear in the training batch images that yolo output during training initialization.
![image](https://github.com/ultralytics/ultralytics/assets/62246708/1357c7b4-9d99-4b99-b1ea-6cab815199ca)

**Comment by glenn-jocher** [comment]
@aymanaboghonim hello,

Thank you for bringing this issue to our attention. From the training batch image you provided, it seems that instances of buildings are not being visualized, which likely indicates a problem in how the data is being processed before or during training.

The first step to debug this issue is to carefully check your dataset, especially the annotation files associated with the images where buildings are expected to appear. Verify that the annotation files correctly match the format expected by YOLOv8 for instance segmentation tasks. Annotations should include not only the bounding boxes but also the object instance masks.

Also, ensure that the classes defined in your `data.yml` file correspond exactly to the classes present in your annotation files, and that class IDs are consistently used throughout the dataset.

After double-checking your annotations and class definitions, I suggest running a small batch of the training process with debug mode enabled, if available, to get more verbose output that might shine a light on what's going wrong.

If everything seems correct with your annotations and class definitions, but you still face the issue, it might be helpful to examine the data loading and augmentation part of the code to make sure there are no issues with how the data is being prepared for training.

Lastly, if this problem persists after taking the above steps, consider isolating a few annotated images with buildings and try training with a significantly reduced dataset to see if the instances appear correctly during training initialization.

Remember to always back up your dataset and configuration files before making any changes, to avoid any data loss.

I hope this helps you move forward with your training process. Should you have any more details or updates regarding this issue, please feel free to share, and we can look into this further.

Best regards,
The Ultralytics Team

**Comment by aymanaboghonim** [comment]
Sorry for late reply. 
Thanks for your prompt answer.
It was issue with annotation files actually.

**Comment by glenn-jocher** [comment]
@aymanaboghonim No problem at all, I'm glad to hear that you've identified the issue with the annotation files. Proper annotations are crucial for the successful training of object detection and segmentation models, so resolving problems at this step is a significant step forward.

If you've corrected the annotations and your training is now proceeding as expected, that's excellent news! Should you encounter any further issues or have more questions as you continue to work with YOLOv8, please don't hesitate to reach out.

Best wishes on your project, and happy training! 🚀
